packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/ClientModelPlugin.cs

```diff
@@ -33,14 +33,14 @@ public class ClientModelPlugin : CodeModelPlugin
         /// Returns the serialization type providers for the given input type.
         /// </summary>
         /// <param name="inputType">The input type.</param>
-        public override IReadOnlyList<TypeProvider> GetSerializationTypeProviders(InputType inputType)
+        protected override IReadOnlyList<TypeProvider> CreateSerializationsCore(InputType inputType)
         {
             switch (inputType)
             {
                 case InputModelType inputModel when inputModel.Usage.HasFlag(InputModelTypeUsage.Json):
                     return [new MrwSerializationTypeDefinition(inputModel)];
                 default:
-                    return base.GetSerializationTypeProviders(inputType);
+                    return base.CreateSerializationsCore(inputType);
```

Comment:
we have to cache the serialization result, therefore I changed this anyway. But I did not move it to the `TypeFactory` class in this PR